pyquil/api/_qam.py

```diff
@@ -75,6 +76,22 @@ def write_memory(self, *, region_name: str, offset: int = 0, value=None):
 
         return self
 
+    @_record_call
+    def write_memory(self, *, region_name: str, values: List[Union[int, float]] = None):
```

Comment:
AFAIK Python does not support this kind of dispatch: this is just overwriting the previous definition of `write_memory()`.